agent/engine/docker_container_engine_test.go

```diff
@@ -355,7 +355,7 @@ func TestCreateContainerTimeout(t *testing.T) {
 		warp <- time.Now()
 		wait.Wait()
 		// Don't return, verify timeout happens
-	})
+	}).MaxTimes(1)
```

Comment:
why was this added?

Reply:
Because this test sometimes fails due to this missing call. Checking the code, this could happen if the `xContainerShortTimeout` is too short, where the `CreateContainer` hasn't been called but the context has returned [here](https://github.com/richardpen/amazon-ecs-agent/blob/master/agent/engine/docker_container_engine.go#L458).

Reply:
This means that sometimes, the code-path where the context might be done while a `createContainer` call is in progress will not be verified by this test. I think the right way to fix this is by wiring in a context to the `CreateContainer` [call](https://github.com/richardpen/amazon-ecs-agent/blob/ea58fddf8462034ff2034aa9788b02341d7702e3/agent/engine/docker_container_engine.go#L440).  Thinking more about it, all of the APIs in the `dockerGoClient` struct need `context`s to be wired in from the caller. Can you please add a TODO and open an issue for that?

Reply:
Issue #1212 was created.